### PR TITLE
Security update: commons-codec

### PR DESCRIPTION
**Pull request by pfistererm**
Updated version of commons-codec (transitive dependency in
licensescout-report-maven-plugin) to 1.14 because of known security
issues with version 1.11

The security alert that triggered this action:
https://snyk.io/vuln/SNYK-JAVA-COMMONSCODEC-561518

The dependency is introduced by:

Detailed paths
Introduced through: org.aposin.licensescout:licensescout-report-maven-plugin@1.4.${revision} › org.apache.maven.doxia:doxia-core@1.9 › org.apache.httpcomponents:httpclient@4.5.8 › commons-codec:commons-codec@1.11
Remediation: No remediation path available.
Introduced through: org.aposin.licensescout:licensescout-report-maven-plugin@1.4.${revision} › org.apache.maven.doxia:doxia-site-renderer@1.9.1 › org.apache.maven.doxia:doxia-core@1.9 › org.apache.httpcomponents:httpclient@4.5.8 › commons-codec:commons-codec@1.11
Remediation: No remediation path available.
Introduced through: org.aposin.licensescout:licensescout-report-maven-plugin@1.4.${revision} › org.apache.maven.reporting:maven-reporting-impl@3.0.0 › org.apache.maven.doxia:doxia-core@1.9 › org.apache.httpcomponents:httpclient@4.5.8 › commons-codec:commons-codec@1.11
Remediation: No remediation path available.
Introduced through: org.aposin.licensescout:licensescout-report-maven-plugin@1.4.${revision} › org.apache.maven.reporting:maven-reporting-impl@3.0.0 › org.apache.maven.doxia:doxia-site-renderer@1.9.1 › org.apache.maven.doxia:doxia-core@1.9 › org.apache.httpcomponents:httpclient@4.5.8 › commons-codec:commons-codec@1.11
Remediation: No remediation path available.

Newer versions of doxia and httpclient still depend on the same version of commons-codec, so locking the version of the transitive dependency is currently the only possible solution.

## Type of request

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring 
- [ ] Documentation or documentation changes

#### Initiator
Matthias Pfisterer

## Checklist:

- [x] All the commits are signed.
- [x] My code follows the code style of this project.
- [x] I agree with die CLA.
- [x] I have read the CONTRIBUTING docs.
- [x] I have added/updated necessary documentation (if appropriate).

### CODE REVIEW

File: licensescout-parent/pom.xml [hunk 1]
```diff
@@ -195,7 +196,16 @@
...
 
 	<dependencyManagement>
+
 		<dependencies>
+
+			<!-- groupId: commons-codec -->
+			<dependency>
+				<groupId>commons-codec</groupId>
+				<artifactId>commons-codec</artifactId>
+				<version>${commons-codec.version}</version>
```
Comment (d-gs) [inline licensescout-parent/pom.xml:206]: This dependency wasn't here before and is not used explicitly anywhere...is it a transitive dependency? If so, could you instead update the other dependency? If not, then add a comment saying that this is a transitive dependency added for security reasons...

Comment (pfistererm) [reply licensescout-parent/pom.xml:206]: Yes, it is a transitive dependency. I added more information on the dependency paths in the description and linked a general description of the security issue.
Conclusion is: "Newer versions of doxia and httpclient still depend on the same version of commons-codec, so locking the version of the transitive dependency is currently the only possible solution."

Comment (d-gs) [reply licensescout-parent/pom.xml:206]: Could you add the reference and the comment to the pom? That was what I meant on the comment, as like that we don't risk to clean "unused" dependencies that are security relevant...

Comment (pfistererm) [reply licensescout-parent/pom.xml:217]: OK, understood. I added a remark in the POM.

**Pull request merged**